tchannel/tornado/peer.py

```diff
@@ -519,7 +519,9 @@ def send(
         # peer, we throw exceptions from retry not NoAvailablePeerError.
         peer = self._choose()
         if not peer:
-            raise NoAvailablePeerError("Can't find available peer.")
+            raise NoAvailablePeerError(
+                "Can't find available peer for " + self.service
```

Comment:
Not sure if this is necessary. Peer are the one we connected to hyperbahn instance. It usually won't be service specific.